service/src/main/java/com/epam/esm/impl/CertificateServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+package com.epam.esm.impl;
+
+import com.epam.esm.CertificateDAO;
+import com.epam.esm.CertificateService;
+import com.epam.esm.TagService;
+import com.epam.esm.dto.CertificateDTO;
+import com.epam.esm.dto.TagDTO;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entitydtomapper.CertificateDtoMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CertificateServiceImpl implements CertificateService {
+
+    private CertificateDAO certificateDAO;
+    @Autowired
+    private TagService tagService;
+    private CertificateDtoMapper certificateDtoMapper;
+
+    @Autowired
+    public CertificateServiceImpl(CertificateDAO certificateDAO, CertificateDtoMapper certificateDtoMapper) {
+        this.certificateDAO = certificateDAO;
+        this.certificateDtoMapper = certificateDtoMapper;
+    }
+
+    @Override
+    public List<CertificateDTO> getAllCertificates() {
```

Comment:
Please use @Transactiona(readOnly = true) annotation for read operations too.
![image](https://user-images.githubusercontent.com/32880986/104951425-84728000-59d3-11eb-9a99-709ae0991f0c.png)
https://stackoverflow.com/questions/44984781/what-are-advantages-of-using-transactionalreadonly-true